govcd/org.go

```diff
@@ -233,6 +235,100 @@ func (adminOrg *AdminOrg) GetVdcByName(vdcname string) (Vdc, error) {
+	}
+	if vdcDefinition.ComputeCapacity[0] == nil {
+		return errors.New("VdcConfiguration missing required field: ComputeCapacity[0]")
+	}
+	if vdcDefinition.ComputeCapacity[0].CPU == nil {
+		return errors.New("VdcConfiguration missing required field: ComputeCapacity[0].CPU")
+	}
+	if vdcDefinition.ComputeCapacity[0].CPU.Units == "" {
+		return errors.New("VdcConfiguration missing required field: ComputeCapacity[0].CPU.Units")
+	}
+	if vdcDefinition.ComputeCapacity[0].Memory == nil {
+		return errors.New("VdcConfiguration missing required field: ComputeCapacity[0].Memory")
+	}
+	if vdcDefinition.ComputeCapacity[0].Memory.Units == "" {
+		return errors.New("VdcConfiguration missing required field: ComputeCapacity[0].Memory.Units")
+	}
+	if vdcDefinition.VdcStorageProfile == nil {
+		return errors.New("VdcConfiguration missing required field: VdcStorageProfile")
+	}
+	if vdcDefinition.VdcStorageProfile.Units == "" {
+		return errors.New("VdcConfiguration missing required field: VdcStorageProfile.Units")
+	}
+	if vdcDefinition.ProviderVdcReference == nil {
+		return errors.New("VdcConfiguration missing required field: ProviderVdcReference")
+	}
+	if vdcDefinition.ProviderVdcReference.HREF == "" {
+		return errors.New("VdcConfiguration missing required field: ProviderVdcReference.HREF")
+	}
+	return nil
+}
+
+// CreateVdc creates a VDC with the given params under the given organization.
+// Returns an AdminVdc.
+// API Documentation: https://code.vmware.com/apis/220/vcloud#/doc/doc/operations/POST-VdcConfiguration.html
+func (org *AdminOrg) CreateVdc(vdcConfiguration *types.VdcConfiguration) (AdminVdc, error) {
+	err := validateVdcConfiguration(vdcConfiguration)
+	if err != nil {
+		return AdminVdc{}, err
+	}
+	output, err := xml.MarshalIndent(vdcConfiguration, "  ", "    ")
+	if err != nil {
+		return AdminVdc{}, fmt.Errorf("error marshalling xml: %s", err)
+	}
+	xmlData := bytes.NewBufferString(xml.Header + string(output))
+	util.Logger.Printf("[TRACE] AdminOrg.CreateVdc - xml payload: %s\n", xmlData)
+	vdcCreateHREF, err := url.ParseRequestURI(org.AdminOrg.HREF)
+	if err != nil {
+		return AdminVdc{}, fmt.Errorf("error parsing admin org url: %s", err)
+	}
+	vdcCreateHREF.Path += "/vdcsparams"
+	req := org.client.NewRequest(map[string]string{}, "POST", *vdcCreateHREF, xmlData)
+	req.Header.Add("Content-Type", "application/vnd.vmware.admin.createVdcParams+xml")
+	resp, err := checkResp(org.client.Http.Do(req))
+	if err != nil {
+		return AdminVdc{}, fmt.Errorf("error instantiating a new VDC: %s", err)
+	}
+
+	adminVdc := NewAdminVdc(org.client)
+	if err = decodeBody(resp, adminVdc.AdminVdc); err != nil {
+		return AdminVdc{}, fmt.Errorf("error decoding admin VDC response: %s", err)
+	}
+	return *adminVdc, nil
+}
+
+// Creates the vdc and waits for the asynchronous task to complete.
+func (org *AdminOrg) CreateVdcWait(vdcDefinition *types.VdcConfiguration) (AdminVdc, error) {
```

Comment:
The type `AdminVdc` is not visible without being qualified with the `types` package.